Add option to disable building/running tests with assembly code dependencies

Attempting to create a Fedora package for OpenCoarrays but I'm getting:
```
/usr/bin/gfortran   -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -I/usr/lib64/gfortran/modules/mpich -DNDEBUG CMakeFiles/hello_multiverse.dir/hello_multiverse.F90.o  -o hello_multiverse  ../../../mpi/libcaf_mpi.a 
../../../mpi/libcaf_mpi.a(mpi_caf.c.o): In function `error_stop':
/builddir/build/BUILD/OpenCoarrays-1.7.4/src/mpi/mpi_caf.c:3776: undefined reference to `MPI_Abort'
```

Full build log:
[build.txt](https://github.com/sourceryinstitute/opencoarrays/files/577213/build.txt)

It seems that the binary isn't being linked to the mpi library.

Ah, I see the error now - I wasn't actually passing in CC=mpicc and FC=mpif90 although I thought I was.

Now I get:
```
/usr/lib64/mpich/bin/mpif90   -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fs
tack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -I/usr/lib64/gfortran/modules/mpich
 -DNDEBUG CMakeFiles/coarray_navier_stokes.dir/coarray-shear_coll.F90.o  -o coarray_navier_stokes  ../../../../../../src/tests/integration/pde_solvers/navier-stokes/libfft_
sse.a ../../../../mpi/libcaf_mpi.a
make[2]: Leaving directory '/builddir/build/BUILD/OpenCoarrays-1.7.4/build-mpich'
/usr/bin/ld: ../../../../../../src/tests/integration/pde_solvers/navier-stokes/libfft_sse.a(trig.o): relocation R_X86_64_32 against `.rodata' can not be used when making a
shared object; recompile with -fPIC
/usr/bin/ld: ../../../../../../src/tests/integration/pde_solvers/navier-stokes/libfft_sse.a(plan.o): relocation R_X86_64_32 against `.rodata.str1.4' can not be used when ma
king a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -209,7 +209,7 @@ if(opencoarrays_aware_compiler)
   add_mpi_test(co_reduce 4 ${tests_root}/unit/collectives/co_reduce_test)
   add_mpi_test(syncimages_status 32 ${tests_root}/unit/sync/syncimages_status)
   add_mpi_test(simpleatomics 32 ${tests_root}/unit/simple/atomics)
-  # I think there is a logic error in the following test
+  # possible logic error in the following test
 #  add_mpi_test(increment_my_neighbor 32 ${tests_root}/unit/simple/increment_my_neighbor)
 
   # Integration tests verifying the use of libcaf_mpi in applications
@@ -218,7 +218,7 @@ if(opencoarrays_aware_compiler)
   add_mpi_test(co_heat 2 ${tests_root}/integration/pde_solvers/coarrayHeatSimplified/co_heat)
   add_mpi_test(asynchronous_hello_world 2 ${tests_root}/integration/events/asynchronous_hello_world)
   if ( ("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "x86_64") AND ("${CMAKE_SYSTEM_NAME}" MATCHES "Linux") )
-    if ( NOT (DEFINED ENV{TRAVIS}))
+    if ( (NOT (DEFINED ENV{TRAVIS})) AND (NOT SKIP_ASSEMBLY_DEPS) )
       add_mpi_test(coarray_navier_stokes 2 ${tests_root}/integration/pde_solvers/navier-stokes/coarray_navier_stokes)
       set_property(TEST coarray_navier_stokes PROPERTY PASS_REGULAR_EXPRESSION "Test passed.")
     endif()

--- a/src/tests/CMakeLists.txt
+++ b/src/tests/CMakeLists.txt
@@ -1,6 +1,9 @@
 if ("${CMAKE_SYSTEM_NAME}" MATCHES "Linux")
   option(HIGH_RESOLUTION_TIMER "use the walltime.o wallclock timer" FALSE)
   option(LEGACY_ARCHITECTURE "use the SSE-enabled FFT library" FALSE)
+
+  # Let package maintainers disable fft library written in assembly
+  option ( SKIP_ASSEMBLY_DEPS "Skip buidling/running tests that depend on files written in assembly" FALSE )
 endif()
 
 if (HIGH_RESOLUTION_TIMER)

--- a/src/tests/integration/pde_solvers/navier-stokes/CMakeLists.txt
+++ b/src/tests/integration/pde_solvers/navier-stokes/CMakeLists.txt
@@ -1,7 +1,7 @@
 # checking whether the machine is of type 64-bit before proceeding further
 if ( ("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "x86_64") AND ("${CMAKE_SYSTEM_NAME}" MATCHES "Linux") )
   # Default to older SSE-instruction-based FFT library
-  if (NOT (DEFINED ENV{TRAVIS}))
+  if (NOT (DEFINED ENV{TRAVIS}) AND (NOT SKIP_ASSEMBLY_DEPS) )
     if (LEGACY_ARCHITECTURE OR (NOT DEFINED(LEGACY_ARCHITECTURE)))
       set(fft_library ${CMAKE_CURRENT_SOURCE_DIR}/libfft_sse.a )
     else()

